### PR TITLE
Add test for #11747, TFrame double deletion

### PR DESCRIPTION
**Pull request by pcanal**

### CODE REVIEW

File: root/io/pointers/restoreCanvas.C [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+   // set to not use RecursiveRemove.  In order to detect memory leak in that list
+   // we need to turn it on .. but doing so, also prevents the double delete,
+   // so we need to also remove the Canvas from the list of cleaups (indirectly)
+   for(TObject *o : *c->GetListOfPrimitives())
+      o->SetBit(kMustCleanup);
+   if (draw) {
+      c->Draw();
+      c->Update();
+   }
+   gROOT->GetListOfCanvases()->Remove(c.get());
```
Comment (Axel-Naumann) [inline root/io/pointers/restoreCanvas.C:90]: Can you verify that this actually found and removed c?

File: root/io/pointers/restoreCanvas.C [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      Info("readdata", "There is no frame");
+   }
+   if (f && f->TestBit(kCanDelete)) {
+      Warning("readdata", "The frame is set as 'kCanDelete'");
+   }
+   if (f) 
+      f->SetBit(kMustCleanup);
+   tcd.fValueMonitored = f;
+   // This was crashing in some cases.
+   c.reset(); // force delete so that we can check the result below
```
Comment (Axel-Naumann) [inline root/io/pointers/restoreCanvas.C:103]: Can you check what valgrind says, is this now free of memory errors?

File: root/io/pointers/restoreCanvas.C [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+   }
+   return 0;
+}
+
+
+int restoreCanvas()
+{
+   TMemFile f("withcanvas.root", "NEW");
+   writedata(f);
+   //TFile f("hsimplecanvas.root");
```
Comment (Axel-Naumann) [inline root/io/pointers/restoreCanvas.C:117]: I don't understand the motivation for keeping these two commented lines?

File: root/io/pointers/restoreCanvas.C [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+#include "TH1.h"
+#include "TError.h"
+#include "TROOT.h"
+
+void writedata(TFile &file)
+{
+   auto h = new TH1F("h", "", 100, 0, 10);
+   h->FillRandom("gaus");
+
+   auto c = new TCanvas("c1");
```
Comment (Axel-Naumann) [inline root/io/pointers/restoreCanvas.C:13]: Can you rename c1..c3 to something more telling, like "frameThatGetsDeleted" etc? I'm afraid of the complexity of this test: loops, classes, output that's sometimes expected and sometimes not, etc. Not a lot of fun to debug.

**Review by Axel-Naumann — APPROVED** [review]
But see comments

File: root/io/pointers/restoreCanvas.C (hunk 3 above)
Comment (pcanal) [reply root/io/pointers/restoreCanvas.C:117]: oversight.

File: root/io/pointers/restoreCanvas.C (hunk 4 above)
Comment (pcanal) [reply root/io/pointers/restoreCanvas.C:13]: I renamed the canvas.  Unfortunately, I don't see a much better way to test all the cases.

File: root/io/pointers/restoreCanvas.C (hunk 1 above)
Comment (pcanal) [reply root/io/pointers/restoreCanvas.C:90]: ok.  I added:
```
   if (gROOT->GetListOfCanvases()->FindObject(c.get())) {
      Error("readdata", "Failed to remove the canvas '%s' from the list of canvases", name);
      return 5;
   }
```

File: root/io/pointers/restoreCanvas.C (hunk 2 above)
Comment (pcanal) [reply root/io/pointers/restoreCanvas.C:103]: Of course, it was clean.  I double check again :)

**Comment by pcanal** [comment]
The error on `ROOT-ubuntu2004/python3` are due to the CI taking the roottest branch verbatim instead of rebasing on master (i.e. missing roottest updates)

**Comment by Axel-Naumann** [comment]
The remaining failures are pre-existing.

**Review by couet — APPROVED** [review]

**Pull request merged**